src/device.hpp

```diff
@@ -63,12 +63,12 @@ typedef struct _cl_device_id {
     }
 
     uint64_t memory_size() const {
-        return std::min(max_alloc_size() * 4, actual_memory_size());
+        return std::min<uint64_t>(max_alloc_size() * 4, actual_memory_size());
     }
 
     size_t max_alloc_size() const {
         size_t max_buffer_size = m_properties.limits.maxStorageBufferRange;
-        return std::min(max_buffer_size, actual_memory_size());
+        return std::min<uint64_t>(max_buffer_size, actual_memory_size());
```

Comment:
You can get rid of this one by changing the type of `max_buffer_size` to `uint64_t`; `size_t` was a kind of arbitrary and not very helpful choice on my part.